src/shared/cloudformation/activation.ts

```diff
@@ -17,8 +19,18 @@ export const TEMPLATE_FILE_GLOB_PATTERN = '**/template.{yaml,yml}'
  * @param extensionContext VS Code extension context
  */
 export async function activate(extensionContext: vscode.ExtensionContext): Promise<void> {
-    const registry = CloudFormationTemplateRegistry.getRegistry()
-    const manager = new CloudFormationTemplateRegistryManager(registry)
-    await manager.addTemplateGlob(TEMPLATE_FILE_GLOB_PATTERN)
-    extensionContext.subscriptions.push(manager)
+    try {
+        const registry = CloudFormationTemplateRegistry.getRegistry()
+        const manager = new CloudFormationTemplateRegistryManager(registry)
+        await manager.addTemplateGlob(TEMPLATE_FILE_GLOB_PATTERN)
+        extensionContext.subscriptions.push(manager)
+    } catch (e) {
+        vscode.window.showErrorMessage(
+            localize(
+                'AWS.codelens.failToInitialize',
+                'Failed to activate template registry. CodeLenses will not appear on SAM template files.'
+            )
+        )
+        getLogger().error('Activating the template registry failed', e)
```

Comment:
As a general rule it is useful to avoid different wording (in all contexts) unless there is an actual reason to have different wording. Avoids unnecessary entropy.

```suggestion
        getLogger().error('Failed to activate template registry', e)
```